README.md

```diff
@@ -1,44 +1,428 @@
+
+The code that is allowed in `runJS` is designed to work similarly to the [TC39 Module Blocks Proposal](https://github.com/tc39/proposal-js-module-blocks), and eventually we hope to be able to switch to that official syntax.
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'runJS example',
+  withBrowser(async ({ utils }) => {
+    await utils.runJS(`
+      // ./other-file is resolved from the test file that called `runJS`
+      import { render } from './other-file'
+      // top-level await is supported
+      await render()
+    `);
+  }),
+);
+```
+
+#### `TestMuleUtils.loadJS(jsPath: string): Promise<void>`
+
+Load a JS (or TS, JSX) file into the browser. Pass a path that will be resolved from your test file.
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'loadJS example',
+  withBrowser(async ({ utils }) => {
+    await utils.loadJS('./button');
+  }),
+);
+```
+
+#### `TestMuleUtils.loadCSS(cssPath: string): Promise<void>`
+
+Load a CSS (or Sass, Less, etc.) file into the browser. Pass a path that will be resolved from your test file.
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'loadCSS example',
+  withBrowser(async ({ utils }) => {
+    await utils.loadCSS('./button.sass');
+  }),
+);
+```
+
+#### `TestMuleUtils.injectCSS(css: string): Promise<void>`
+
+Set the contents of a new `<style>` tag.
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'injectCSS example',
+  withBrowser(async ({ utils }) => {
+    await utils.injectCSS(`
+      .button {
+        background: green;
+      }
+    `);
+  }),
+);
+```
+
+#### `TestMuleUtils.injectHTML(html: string): Promise<void>`
+
+Set the contents of `document.body`.
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'injectHTML example',
+  withBrowser(async ({ utils }) => {
+    await utils.injectHTML(`
+      <h1>Hi</h1>
+    `);
+  }),
+);
+```
+
+### [`jest-dom`](https://github.com/testing-library/jest-dom) Matchers
+
+Test Mule adds [`jest-dom`'s matchers](https://github.com/testing-library/jest-dom#table-of-contents) to Jest's `expect` global. They are slightly modified from the original matchers, they are wrapped to execute in the browser, and return a Promise.
+
+**Don't forget to `await` matchers! This is necessary because the matchers execute in the browser. If you forget, your matchers may execute after your test finishes, and you may get obscure errors.**
+
+```js
+import { withBrowser } from 'test-mule';
+
+test(
+  'jest-dom matchers example',
+  withBrowser(async ({ screen }) => {
+    const button = await screen.getByText();
+    // jest-dom matcher -- Runs in browser, *must* be awaited
+    await expect(button).toBeVisible();
+    // Built-in Jest matcher -- Runs only in Node, does not need to be awaited
+    expect(5).toEqual(5);
+  }),
+);
+```
+
+## Comparisons with other testing tools/Why does this exist?
```

Comment:
This is great. And this location feels about right for this detailed look at the alternatives. I feel like we should also split off a 1-3 sentence version of answering the "Why?" way up at the top.